uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/file/popups/RenamePopUpPresenter.java

```diff
@@ -97,22 +131,40 @@ public void setup() {
     }
 
     public void rename(final String newName) {
+
+        final String fileName = newFileName(newName);
+        final String comment = toggleCommentPresenter.getComment();
+        final ValidatorWithReasonCallback callback = validatorCallback(comment, newName, renameCommand);
+
+        validator.validate(fileName, callback);
+    }
+
+    public void saveAndRename(final String newName) {
+
+        final String fileName = newFileName(newName);
+        final String comment = toggleCommentPresenter.getComment();
+        final ValidatorWithReasonCallback callback = validatorCallback(comment, newName, saveAndRenameCommand);
+
+        validator.validate(fileName, callback);
+    }
+
+    private String newFileName(final String newName) {
+
         final String extension = extension(path.getFileName());
-        final String fileName = newName + extension;
 
-        validator.validate(fileName,
-                           validatorCallback(toggleCommentPresenter.getComment(),
-                                             newName));
+        return newName + extension;
     }
 
     private String extension(final String fileName) {
         return fileName.lastIndexOf(".") > 0 ? fileName.substring(fileName.lastIndexOf(".")) : "";
     }
 
     private ValidatorWithReasonCallback validatorCallback(final String comment,
-                                                          final String baseFileName) {
+                                                          final String baseFileName,
+                                                          final CommandWithFileNameAndCommitMessage onSuccess) {
+
         return new ValidatorWithReasonCallback() {
-            @Override
```

Comment:
Any reason you removed the ```@Override```'s in this anonymous inner class?

Reply:
No. I'm bringing it back.